pkg/db/db.go

```diff
@@ -231,6 +232,8 @@ func deserializeRecord(r *bufio.Reader) (key string, val []byte, seq uint64, err
 	if err = binary.Read(r, binary.LittleEndian, &seq); err != nil {
 		return
 	}
+	repro = isRepro(seq)
+	seq = getSeq(seq)
 	if seq == seqDeleted {
```

Comment:
You just stripped high bit, so now seq will never be equal to seqDeleted which has high bit set.
The bug is triggered with this test:

```go
func TestDeletedRecord(t *testing.T) {
	fn := tempFile(t)
	defer os.Remove(fn)
	db, err := Open(fn)
	if err != nil {
		t.Fatalf("failed to open db: %v", err)
	}
	db.Save("1", []byte("a"), 0, false)
	db.Save("2", []byte("b"), 1, false)
	db.Save("3", []byte("c"), 2, false)
	if err := db.Flush(); err != nil {
		t.Fatalf("failed to flush db: %v", err)
	}
	db.Delete("2")
	if err := db.Flush(); err != nil {
		t.Fatalf("failed to flush db: %v", err)
	}
	db, err = Open(fn)
	if err != nil {
		t.Fatalf("failed to open db: %v", err)
	}
	if len(db.Records) != 2 {
		t.Fatalf("bad record count %v, want 2", len(db.Records))
	}
	for key := range db.Records {
		switch key {
		case "1", "3":
			continue
		default:
			t.Fatalf("unknown key: %v", key)
		}
	}
}
```